specifications/bcastByz/bcastByz.tla

```diff
@@ -0,0 +1,983 @@
+        BY <1>5, <2>5, <3>4 DEFS UponAcceptNotSent, TypeOK, Receive
+    <2>7 QED
+      BY <1>2, <1>5, <2>1, <2>2, <2>3, <2>4, <2>5, <2>6 DEF TypeOK
+  <1>6 CASE Receive(i) /\ UponAccept(i)
+    <2>1 pc' \in [ Proc -> {"V0", "V1", "SE", "AC"} ]
+      BY <1>6 DEFS UponAccept, TypeOK
+    <2>2 Corr' \subseteq Proc'
+      BY <1>6 DEFS UponAccept, TypeOK
+    <2>3 Faulty' \subseteq Proc'
+      BY <1>6 DEFS UponAccept, TypeOK    
+    <2>4 sent' \subseteq Proc' \times M'
+      <3>1 i \in Proc
+        BY <1>6 DEFS TypeOK
+      <3>2 << i, "ECHO">> \in Proc' \times M'
+        BY <1>6, <3>1 DEFS Proc, M 
+      <3>3 { << i, "ECHO">> } \subseteq Proc' \times M'
+        BY <3>2
+      <3>4 sent \subseteq Proc' \times M'
+        BY <1>6 DEFS TypeOK, M, Proc
+      <3>5 QED
+        BY <1>6, <3>3, <3>4 DEFS UponAccept, TypeOK            
+    <2>5 newMessages' \in SUBSET ( sent' \cup ByzMsgs' )
+      <3>1 sent \subseteq sent'
+        BY <1>6 DEFS UponAccept
+      <3>2 Faulty' = Faulty
+        BY <1>6 DEFS UponAccept
+      <3>3 ByzMsgs \subseteq ByzMsgs'
+        BY <1>6, <3>2 DEF ByzMsgs  
+      <3>4 QED
+        BY <1>6, <3>1, <3>3 DEFS Receive, TypeOK         
+    <2>6 rcvd' \in [ Proc -> SUBSET (sent' \cup ByzMsgs') ]    
+      <3>1 Faulty \subseteq Faulty'
+        BY <1>6 DEFS UponAccept
+      <3>2 ByzMsgs \subseteq ByzMsgs'
+        BY <3>1 DEF ByzMsgs
+      <3>3 sent \subseteq sent'
+        BY  <1>6 DEF UponAccept
+      <3>4 (sent \cup ByzMsgs)  \subseteq (sent' \cup ByzMsgs')
+        BY <3>2, <3>3
+      <3>5 QED             
+        BY <1>6, <2>5, <3>4 DEFS UponAccept, TypeOK, Receive
+    <2>7 QED
+      BY <1>2, <1>6, <2>1, <2>2, <2>3, <2>4, <2>5, <2>6 DEF TypeOK  
+  <1>7 CASE Receive(i) /\ UNCHANGED << pc, sent, Corr, Faulty >>
+    <2>1 pc' \in [ Proc -> {"V0", "V1", "SE", "AC"} ]
+      BY <1>7 DEF TypeOK
+    <2>2 Corr' \subseteq Proc'
+      BY <1>7 DEF TypeOK
+    <2>3 Faulty' \subseteq Proc'
+      BY <1>7 DEF TypeOK
+    <2>4 sent' \subseteq Proc' \times M'
+      BY <1>7 DEF TypeOK        
+    <2>5 newMessages' \in SUBSET ( sent' \cup ByzMsgs' )
+      <3>1 sent \subseteq sent'
+        BY <1>7 DEF TypeOK
+      <3>2 Faulty' = Faulty
+        BY <1>7 
+      <3>3 ByzMsgs \subseteq ByzMsgs'
+        BY <1>7, <3>2 DEF ByzMsgs  
+      <3>4 QED
+        BY <1>7, <3>1, <3>3 DEFS Receive, TypeOK         
+    <2>6 rcvd' \in [ Proc -> SUBSET (sent' \cup ByzMsgs') ]   
+      <3>1 Faulty \subseteq Faulty'
+        BY <1>7
+      <3>2 ByzMsgs \subseteq ByzMsgs'
+        BY <3>1 DEF ByzMsgs
+      <3>3 sent \subseteq sent'
+        BY  <1>7
+      <3>4 (sent \cup ByzMsgs)  \subseteq (sent' \cup ByzMsgs')
+        BY <3>2, <3>3
+      <3>5 QED             
+        BY <1>7, <2>5, <3>4 DEFS TypeOK, Receive
+    <2>7 QED
+      BY <1>2, <1>7, <2>1, <2>2, <2>3, <2>4, <2>5, <2>6 DEF TypeOK  
+  <1>8 QED
+    BY <1>1, <1>2, <1>3, <1>4, <1>5, <1>6, <1>7 DEF Step  
+  
+THEOREM TypeOKThm == SpecNoBcast => TypeOK
```

Comment:
Again, the conclusion should be "... => []TypeOK", and step <1>3 should use PTL, otherwise the proof of <1>2 is in fact useless.